pkg/storage/fs/ocis/tree.go

```diff
@@ -0,0 +1,282 @@
+			return errors.Wrap(err, "ocisfs: could not symlink parent")
+		}
+
+		// rename child
+		err = os.Rename(
+			filepath.Join(fs.DataDirectory, "nodes", oldNode.ParentID, "children", oldNode.Name),
+			filepath.Join(fs.DataDirectory, "nodes", oldNode.ParentID, "children", newNode.Name),
+		)
+		if err != nil {
+			return errors.Wrap(err, "ocisfs: could not rename symlink")
+		}
+		return fs.Propagate(ctx, oldNode)
+	}
+	// we are moving the node to a new parent, any target has been removed
+	// bring old node to the new parent
+
+	nodePath := filepath.Join(fs.DataDirectory, "nodes", oldNode.ID)
+
+	// update back link
+	// we are not only linking back to the parent, but also to the filename
+	err = os.Remove(filepath.Join(nodePath, "parentname"))
+	if err != nil {
+		return errors.Wrap(err, "ocisfs: could not remove parent link")
+	}
+	err = os.Symlink("../"+newNode.ParentID+"/children/"+newNode.Name, filepath.Join(nodePath, "parentname"))
+	if err != nil {
+		return errors.Wrap(err, "ocisfs: could not symlink parent")
+	}
+
+	// rename child
+	err = os.Rename(
+		filepath.Join(fs.DataDirectory, "nodes", oldNode.ParentID, "children", oldNode.Name),
+		filepath.Join(fs.DataDirectory, "nodes", newNode.ParentID, "children", newNode.Name),
+	)
+	if err != nil {
+		return errors.Wrap(err, "ocisfs: could not rename symlink")
+	}
+
+	// TODO inefficient because we might update several nodes twice, only propagate unchanged nodes?
+	// collect in a list, then only stat each node once
+	// also do this in a go routine ... webdav should check the etag async
+	err = fs.Propagate(ctx, oldNode)
+	if err != nil {
+		return errors.Wrap(err, "ocisfs: Move: could not propagate old node")
+	}
+	err = fs.Propagate(ctx, newNode)
+	if err != nil {
+		return errors.Wrap(err, "ocisfs: Move: could not propagate old node")
+	}
+	return nil
+}
+
+// ChildrenPath returns the absolute path to childrens in a node
+// TODO move to node?
+func (fs *Tree) ChildrenPath(node *NodeInfo) string {
+	return filepath.Join(fs.DataDirectory, "nodes", node.ID, "children")
+}
+
+// ListFolder lists the children inside a folder
+func (fs *Tree) ListFolder(ctx context.Context, node *NodeInfo) ([]*NodeInfo, error) {
+
+	children := fs.ChildrenPath(node)
+	f, err := os.Open(children)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, errtypes.NotFound(children)
+		}
+		return nil, errors.Wrap(err, "tree: error listing "+children)
+	}
+
+	names, err := f.Readdirnames(0)
+	nodes := []*NodeInfo{}
+	for i := range names {
+		link, err := os.Readlink(filepath.Join(children, names[i]))
+		if err != nil {
+			// TODO log
+			continue
+		}
+		n := &NodeInfo{
+			ParentID: node.ID,
+			ID:       filepath.Base(link),
+			Name:     names[i],
+			Exists:   true, // TODO
+		}
+
+		nodes = append(nodes, n)
+	}
+	return nodes, nil
+}
+
+// Delete deletes a node in the tree
+func (fs *Tree) Delete(ctx context.Context, node *NodeInfo) (err error) {
+	err = fs.pw.FillParentAndName(node)
```

Comment:
filling the node feels overkill here as we'd get the name anyway

maybe need a GetName() and GetParent() method so the caller can decide whatever they need

Reply:
using Child(name string) and Parent()